hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/builtin/FixedSizeTypesCodec.java

```diff
@@ -72,15 +72,24 @@ public static byte decodeByte(byte[] buffer, int pos) {
     }
 
     public static void encodeUUID(byte[] buffer, int pos, UUID value) {
+        boolean isNull = value == null;
+        encodeBoolean(buffer, pos, isNull);
+        if (isNull) {
+            return;
+        }
         long mostSigBits = value.getMostSignificantBits();
         long leastSigBits = value.getLeastSignificantBits();
-        encodeLong(buffer, pos, mostSigBits);
-        encodeLong(buffer, pos + LONG_SIZE_IN_BYTES, leastSigBits);
+        encodeLong(buffer, pos + BOOLEAN_SIZE_IN_BYTES, mostSigBits);
+        encodeLong(buffer, pos + BOOLEAN_SIZE_IN_BYTES + LONG_SIZE_IN_BYTES, leastSigBits);
     }
 
     public static UUID decodeUUID(byte[] buffer, int pos) {
-        long mostSigBits = decodeLong(buffer, pos);
-        long leastSigBits = decodeLong(buffer, pos + LONG_SIZE_IN_BYTES);
+        boolean isNull = decodeBoolean(buffer, pos);
```

Comment:
It is possible to pass null UUID to the protocol in the authentication. @sancar said that he will fix it with the ownerless client but for the time being we need some sort of null check. Also, UUIDs are again fixed sized (17 bytes instead of 16). If we saw that the UUID is null, we just skip reading the most and least significant bits but they are there